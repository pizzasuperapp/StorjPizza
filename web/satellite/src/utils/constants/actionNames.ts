// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export const APP_STATE_ACTIONS = {
    TOGGLE_TEAM_MEMBERS: 'toggleAddTeamMembersPopup',
    TOGGLE_ACCOUNT: 'toggleAccountDropdown',
    TOGGLE_SELECT_PROJECT_DROPDOWN: 'toggleSelectProjectDropdown',
    TOGGLE_RESOURCES_DROPDOWN: 'toggleResourcesDropdown',
    TOGGLE_QUICK_START_DROPDOWN: 'toggleQuickStartDropdown',
    TOGGLE_SETTINGS_DROPDOWN: 'toggleSettingsDropdown',
    TOGGLE_EDIT_PROJECT_DROPDOWN: 'toggleEditProjectDropdown',
    TOGGLE_FREE_CREDITS_DROPDOWN: 'toggleFreeCreditsDropdown',
    TOGGLE_AVAILABLE_BALANCE_DROPDOWN: 'toggleAvailableBalanceDropdown',
    TOGGLE_PERIODS_DROPDOWN: 'togglePeriodsDropdown',
    TOGGLE_AG_DATEPICKER_DROPDOWN: 'toggleAGDatepickerDropdown',
    TOGGLE_CHARTS_DATEPICKER_DROPDOWN: 'toggleChartsDatepickerDropdown',
    TOGGLE_BUCKET_NAMES_DROPDOWN: 'toggleBucketNamesDropdown',
    TOGGLE_PERMISSIONS_DROPDOWN: 'togglePermissionsDropdown',
    TOGGLE_SUCCESSFUL_PASSWORD_RESET: 'TOGGLE_SUCCESSFUL_PASSWORD_RESET',
    TOGGLE_SUCCESSFUL_PROJECT_CREATION_POPUP: 'toggleSuccessfulProjectCreationPopup',
    TOGGLE_EDIT_PROFILE_POPUP: 'toggleEditProfilePopup',
    TOGGLE_CHANGE_PASSWORD_POPUP: 'toggleChangePasswordPopup',
    TOGGLE_UPLOAD_CANCEL_POPUP: 'toggleUploadCancelPopup',
    SHOW_SET_DEFAULT_PAYMENT_METHOD_POPUP: 'showSetDefaultPaymentMethodPopup',
    CLOSE_SET_DEFAULT_PAYMENT_METHOD_POPUP: 'closeSetDefaultPaymentMethodPopup',
    SHOW_DELETE_PAYMENT_METHOD_POPUP: 'showDeletePaymentMethodPopup',
    CLOSE_DELETE_PAYMENT_METHOD_POPUP: 'closeDeletePaymentMethodPopup',
    CLOSE_POPUPS: 'closePopups',
    CHANGE_STATE: 'changeFetchState',
    TOGGLE_PAYMENT_SELECTION: 'TOGGLE_PAYMENT_SELECTION',
    SET_SATELLITE_NAME: 'SET_SATELLITE_NAME',
    SET_PARTNERED_SATELLITES: 'SET_PARTNERED_SATELLITES',
    SET_SATELLITE_STATUS: 'SET_SATELLITE_STATUS',
    SET_COUPON_CODE_BILLING_UI_STATUS: 'SET_COUPON_CODE_BILLING_UI_STATUS',
    SET_COUPON_CODE_SIGNUP_UI_STATUS: 'SET_COUPON_CODE_SIGNUP_UI_STATUS',
    SET_PROJECT_DASHBOARD_STATUS: 'SET_PROJECT_DASHBOARD_STATUS',
    SET_OBJECTS_FLOW_STATUS: 'SET_OBJECTS_FLOW_STATUS',
};

export const NOTIFICATION_ACTIONS = {
    SUCCESS: 'success',
    ERROR: 'error',
    NOTIFY: 'notify',
    WARNING: 'WARNING',
    DELETE: 'deleteNotification',
    PAUSE: 'pauseNotification',
    RESUME: 'resumeNotification',
    CLEAR: 'clearNotifications',
};

export const PM_ACTIONS = {
    ADD: 'addProjectMembers',
    DELETE: 'deleteProjectMembers',
    TOGGLE_SELECTION: 'toggleProjectMemberSelection',
    CLEAR_SELECTION: 'clearProjectMemberSelection',
    FETCH: 'fetchProjectMembers',
    CLEAR: 'clearProjectMembers',
    SET_SEARCH_QUERY: 'setProjectMembersSearchQuery',
    SET_SORT_BY: 'setProjectMembersSortingBy',
    SET_SORT_DIRECTION: 'setProjectMembersSortingDirection',
    CLEAR_OFFSET: 'clearProjectMembersOffset',
};
