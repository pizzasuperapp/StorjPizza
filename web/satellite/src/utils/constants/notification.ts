// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export const NOTIFICATION_TYPES = {
    SUCCESS: 'SUCCESS',
    NOTIFICATION: 'NOTIFICATION',
    ERROR: 'ERROR',
    WARNING: 'WARNING',
};
