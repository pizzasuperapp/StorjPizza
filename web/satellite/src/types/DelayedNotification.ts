// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import Vue, {VueConstructor} from 'vue';

import { NOTIFICATION_TYPES } from '@/utils/constants/notification';
import { getId } from '@/utils/idGenerator';

import SuccessIcon from "@/../static/images/notifications/success.svg";
import NotificationIcon from "@/../static/images/notifications/notification.svg";
import ErrorIcon from "@/../static/images/notifications/error.svg";
import WarningIcon from "@/../static/images/notifications/warning.svg";

export class DelayedNotification {
    private readonly successColor: string = '#DBF1D3';
    private readonly errorColor: string = '#FFD4D2';
    private readonly infoColor: string = '#D0E3FE';
    private readonly warningColor: string = '#FCF8E3';
    public readonly id: string;

    private readonly callback: () => void;
    private timerId: ReturnType<typeof setTimeout>;
    private startTime: number;
    private remainingTime: number;

    public readonly type: string;
    public readonly message: string;
    public readonly style: { backgroundColor: string };
    public readonly icon: VueConstructor<Vue>;

    constructor(callback: () => void, type: string, message: string) {
        this.callback = callback;
        this.type = type;
        this.message = message;
        this.id = getId();
        this.remainingTime = 3000;
        this.start();

        // Switch for choosing notification style depends on notification type
        switch (this.type) {
        case NOTIFICATION_TYPES.SUCCESS:
            this.style = { backgroundColor: this.successColor };
            this.icon = SuccessIcon;
            break;

        case NOTIFICATION_TYPES.ERROR:
            this.style = { backgroundColor: this.errorColor };
            this.icon = ErrorIcon;
            break;

        case NOTIFICATION_TYPES.WARNING:
            this.style = { backgroundColor: this.warningColor };
            this.icon = WarningIcon;
            break;

        default:
            this.style = { backgroundColor: this.infoColor };
            this.icon = NotificationIcon;
            break;
        }
    }

    public pause(): void {
        clearTimeout(this.timerId);
        this.remainingTime -= new Date().getMilliseconds() - this.startTime;
    }

    public start(): void {
        this.startTime = new Date().getMilliseconds();
        this.timerId = setTimeout(this.callback, this.remainingTime);
    }
}
