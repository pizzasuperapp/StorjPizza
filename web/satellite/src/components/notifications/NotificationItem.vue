// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div :style="notification.style" class="notification-wrap" :class="{ active: isClassActive }" @mouseover="onMouseOver" @mouseleave="onMouseLeave">
        <div class="notification-wrap__content-area">
            <div class="notification-wrap__content-area__image">
                <component :is="notification.icon" />
            </div>
            <div class="notification-wrap__content-area__message-area">
                <p class="notification-wrap__content-area__message">{{ notification.message }}</p>
                <a
                    v-if="isSupportLinkMentioned"
                    :href="requestUrl"
                    class="notification-wrap__content-area__link"
                    target="_blank"
                >
                    {{ requestUrl }}
                </a>
            </div>
        </div>
        <div class="notification-wrap__buttons-group" @click="onCloseClick">
            <span class="notification-wrap__buttons-group__close">
                <CloseIcon />
            </span>
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import CloseIcon from '@/../static/images/notifications/close.svg';

import { DelayedNotification } from '@/types/DelayedNotification';
import { NOTIFICATION_ACTIONS } from '@/utils/constants/actionNames';
import { MetaUtils } from "@/utils/meta";

// @vue/component
@Component({
    components: {
        CloseIcon,
    },
})
export default class NotificationItem extends Vue {
    @Prop({default: () => new DelayedNotification(() => { return; }, '', '')})
    private notification: DelayedNotification;

    public readonly requestUrl = MetaUtils.getMetaContent('general-request-url');
    public isClassActive = false;

    /**
     * Indicates if support word is mentioned in message.
     * Temporal solution, can be changed later.
     */
    public get isSupportLinkMentioned(): boolean {
        return this.notification.message.toLowerCase().includes('support');
    }

    /**
     * Forces notification deletion.
     */
    public onCloseClick(): void {
        this.$store.dispatch(NOTIFICATION_ACTIONS.DELETE, this.notification.id);
    }

    /**
     * Forces notification to stay on page on mouse over it.
     */
    public onMouseOver(): void {
        this.$store.dispatch(NOTIFICATION_ACTIONS.PAUSE, this.notification.id);
    }

    /**
     * Resume notification flow when mouse leaves notification.
     */
    public onMouseLeave(): void {
        this.$store.dispatch(NOTIFICATION_ACTIONS.RESUME, this.notification.id);
    }

    /**
     * Uses for class change for animation.
     */
    public mounted(): void {
        setTimeout(() => {
            this.isClassActive = true;
        }, 100);
    }
}
</script>

<style scoped lang="scss">
    .notification-wrap {
        position: relative;
        right: -100%;
        width: calc(100% - 40px);
        height: auto;
        display: flex;
        justify-content: space-between;
        padding: 20px;
        align-items: center;
        border-radius: 12px;
        margin-bottom: 7px;
        transition: all 0.3s;

        &__content-area {
            display: flex;
            align-items: center;
            font-family: 'font_medium', sans-serif;
            font-size: 14px;

            &__image {
                max-height: 40px;
            }

            &__message-area {
                display: flex;
                flex-direction: column;
                align-items: flex-start;
                justify-content: space-between;
                margin: 0 0 0 17px;
            }

            &__message,
            &__link {
                height: auto;
                width: 270px;
                word-break: break-word;
            }

            &__link {
                margin-top: 5px;
                color: #224ca5;
                text-decoration: underline;
                cursor: pointer;
                word-break: normal;
            }
        }

        &__buttons-group {
            display: flex;

            &__close {
                width: 32px;
                height: 32px;
                cursor: pointer;
            }
        }
    }

    .active {
        right: 0;
    }
</style>
