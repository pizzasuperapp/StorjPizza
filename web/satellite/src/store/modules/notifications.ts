// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { DelayedNotification } from '@/types/DelayedNotification';
import { NOTIFICATION_ACTIONS } from '@/utils/constants/actionNames';
import { NOTIFICATION_TYPES } from '@/utils/constants/notification';

import { NOTIFICATION_MUTATIONS } from '../mutationConstants';
import {StoreModule} from "@/types/store";

export class NotificationsState {
    public notificationQueue: DelayedNotification[] = [];
}

interface NotificationsContext {
    state: NotificationsState
    commit: (string, ...unknown) => void
}

export function makeNotificationsModule(): StoreModule<NotificationsState, NotificationsContext> {
    return {
        state: new NotificationsState(),
        mutations: {
            // Mutation for adding notification to queue
            [NOTIFICATION_MUTATIONS.ADD](state: NotificationsState, notification: DelayedNotification): void {
                state.notificationQueue.push(notification);
            },
            // Mutation for deleting notification to queue
            [NOTIFICATION_MUTATIONS.DELETE](state: NotificationsState, id: string): void {
                if (state.notificationQueue.length < 1) {
                    return;
                }

                const selectedNotification =  getNotificationById(state.notificationQueue, id);
                if (selectedNotification) {
                    selectedNotification.pause();
                    state.notificationQueue.splice(state.notificationQueue.indexOf(selectedNotification), 1);
                }
            },
            [NOTIFICATION_MUTATIONS.PAUSE](state: NotificationsState, id: string): void {
                const selectedNotification =  getNotificationById(state.notificationQueue, id);
                if (selectedNotification) {
                    selectedNotification.pause();
                }
            },
            [NOTIFICATION_MUTATIONS.RESUME](state: NotificationsState, id: string): void {
                const selectedNotification =  getNotificationById(state.notificationQueue, id);
                if (selectedNotification) {
                    selectedNotification.start();
                }
            },
            [NOTIFICATION_MUTATIONS.CLEAR](state: NotificationsState): void {
                state.notificationQueue = [];
            },
        },
        actions: {
            // Commits mutation for adding success notification
            [NOTIFICATION_ACTIONS.SUCCESS]: function ({commit}: NotificationsContext, message: string): void {
                const notification = new DelayedNotification(
                    () => commit(NOTIFICATION_MUTATIONS.DELETE, notification.id),
                    NOTIFICATION_TYPES.SUCCESS,
                    message,
                );

                commit(NOTIFICATION_MUTATIONS.ADD, notification);
            },
            // Commits mutation for adding info notification
            [NOTIFICATION_ACTIONS.NOTIFY]: function ({commit}: NotificationsContext, message: string): void {

                const notification = new DelayedNotification(
                    () => commit(NOTIFICATION_MUTATIONS.DELETE, notification.id),
                    NOTIFICATION_TYPES.NOTIFICATION,
                    message,
                );

                commit(NOTIFICATION_MUTATIONS.ADD, notification);
            },
            // Commits mutation for adding error notification
            [NOTIFICATION_ACTIONS.ERROR]: function ({commit}: NotificationsContext, message: string): void {
                const notification = new DelayedNotification(
                    () => commit(NOTIFICATION_MUTATIONS.DELETE, notification.id),
                    NOTIFICATION_TYPES.ERROR,
                    message,
                );

                commit(NOTIFICATION_MUTATIONS.ADD, notification);
            },
            // Commits mutation for adding error notification
            [NOTIFICATION_ACTIONS.WARNING]: function ({commit}: NotificationsContext, message: string): void {
                const notification = new DelayedNotification(
                    () => commit(NOTIFICATION_MUTATIONS.DELETE, notification.id),
                    NOTIFICATION_TYPES.WARNING,
                    message,
                );

                commit(NOTIFICATION_MUTATIONS.ADD, notification);
            },
            [NOTIFICATION_ACTIONS.DELETE]: function ({commit}: NotificationsContext, id: string): void {
                commit(NOTIFICATION_MUTATIONS.DELETE, id);
            },
            [NOTIFICATION_ACTIONS.PAUSE]: function ({commit}: NotificationsContext, id: string): void {
                commit(NOTIFICATION_MUTATIONS.PAUSE, id);
            },
            [NOTIFICATION_ACTIONS.RESUME]: function ({commit}: NotificationsContext, id: string): void {
                commit(NOTIFICATION_MUTATIONS.RESUME, id);
            },
            [NOTIFICATION_ACTIONS.CLEAR]: function ({commit}): void {
                commit(NOTIFICATION_MUTATIONS.CLEAR);
            },
        },
        getters: {},
    };
}

function getNotificationById(notifications: DelayedNotification[], id: string): DelayedNotification | undefined {
    return notifications.find((notification: DelayedNotification) => notification.id === id);
}
