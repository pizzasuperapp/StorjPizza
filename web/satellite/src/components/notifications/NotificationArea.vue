// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div v-if="doNotificationsExist" class="notification-container">
        <NotificationItem
            v-for="notification in notifications"
            :key="notification.id"
            :notification="notification"
        />
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import NotificationItem from '@/components/notifications/NotificationItem.vue';

import { DelayedNotification } from '@/types/DelayedNotification';

// @vue/component
@Component({
    components: {
        NotificationItem,
    },
})
export default class NotificationArea extends Vue {
    /**
     * Returns all notification queue from store.
     */
    public get notifications(): DelayedNotification[] {
        return this.$store.state.notificationsModule.notificationQueue;
    }

    /**
     * Indicates if any notifications are in queue.
     */
    public get doNotificationsExist(): boolean {
        return this.notifications.length > 0;
    }
}
</script>

<style scoped lang="scss">
    .notification-container {
        width: 417px;
        background-color: transparent;
        display: flex;
        flex-direction: column;
        position: fixed;
        top: 114px;
        right: 17px;
        align-items: flex-end;
        justify-content: space-between;
        border-radius: 12px;
        z-index: 9999;
        overflow: hidden;
    }
</style>
