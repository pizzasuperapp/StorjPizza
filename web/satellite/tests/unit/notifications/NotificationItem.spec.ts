// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import Vuex from 'vuex';

import NotificationItem from '@/components/notifications/NotificationItem.vue';

import { makeNotificationsModule } from '@/store/modules/notifications';
import { DelayedNotification } from '@/types/DelayedNotification';
import { NOTIFICATION_ACTIONS } from '@/utils/constants/actionNames';
import { NOTIFICATION_TYPES } from '@/utils/constants/notification';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();

localVue.use(Vuex);
const pauseSpy = jest.fn();
const resumeSpy = jest.fn();
const deleteSpy = jest.fn();
const notificationModule = makeNotificationsModule();
notificationModule.actions[NOTIFICATION_ACTIONS.PAUSE] = pauseSpy;
notificationModule.actions[NOTIFICATION_ACTIONS.RESUME] = resumeSpy;
notificationModule.actions[NOTIFICATION_ACTIONS.DELETE] = deleteSpy;

const store = new Vuex.Store(notificationModule);

describe('NotificationItem', () => {

    it('renders correctly', () => {
        const wrapper = mount(NotificationItem);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with success props', () => {
        const testMessage = 'testMessage';

        const wrapper = mount(NotificationItem, {
            propsData: {
                notification: new DelayedNotification(
                    jest.fn(),
                    NOTIFICATION_TYPES.SUCCESS,
                    testMessage,
                ),
            },
        });

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.notification-wrap__content-area__message').text()).toMatch(testMessage);
    });

    it('renders correctly with error props', () => {
        const testMessage = 'testMessage';

        const wrapper = mount(NotificationItem, {
            propsData: {
                notification: new DelayedNotification(
                    jest.fn(),
                    NOTIFICATION_TYPES.ERROR,
                    testMessage,
                ),
            },
        });

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.notification-wrap__content-area__message').text()).toMatch(testMessage);
    });

    it('renders correctly with notification props', () => {
        const testMessage = 'testMessage';

        const wrapper = mount(NotificationItem, {
            propsData: {
                notification: new DelayedNotification(
                    jest.fn(),
                    NOTIFICATION_TYPES.NOTIFICATION,
                    testMessage,
                ),
            },
        });

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.notification-wrap__content-area__message').text()).toMatch(testMessage);
    });

    it('renders correctly with warning props', () => {
        const testMessage = 'testMessage';

        const wrapper = mount(NotificationItem, {
            propsData: {
                notification: new DelayedNotification(
                    jest.fn(),
                    NOTIFICATION_TYPES.WARNING,
                    testMessage,
                ),
            },
        });

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.notification-wrap__content-area__message').text()).toMatch(testMessage);
    });

    it('renders correctly with support link', async (): Promise<void> => {
        const testMessage = 'testMessage support rest of message';

        const wrapper = mount(NotificationItem, {
            propsData: {
                notification: new DelayedNotification(
                    jest.fn(),
                    NOTIFICATION_TYPES.NOTIFICATION,
                    testMessage,
                ),
            },
        });

        await wrapper.setData({ requestUrl: 'https://requestUrl.com' })

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.notification-wrap__content-area__message').text()).toMatch(testMessage);
        expect(wrapper.find('.notification-wrap__content-area__link').text()).toBeTruthy();
    });

    it('trigger pause correctly', () => {
        const wrapper = mount(NotificationItem, { store, localVue });

        wrapper.find('.notification-wrap').trigger('mouseover');

        expect(pauseSpy).toHaveBeenCalledTimes(1);
    });

    it('trigger resume correctly', () => {
        const wrapper = mount(NotificationItem, { store, localVue });

        wrapper.find('.notification-wrap').trigger('mouseover');
        wrapper.find('.notification-wrap').trigger('mouseleave');

        expect(resumeSpy).toHaveBeenCalledTimes(1);
    });

    it('trigger delete correctly', () => {
        const wrapper = mount(NotificationItem, { store, localVue });

        wrapper.find('.notification-wrap__buttons-group').trigger('click');

        expect(deleteSpy).toHaveBeenCalledTimes(1);
    });
});
