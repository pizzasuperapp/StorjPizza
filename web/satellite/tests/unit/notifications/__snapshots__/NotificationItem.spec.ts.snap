// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationItem renders correctly 1`] = `
<div class="notification-wrap" style="background-color: rgb(208, 227, 254);">
  <div class="notification-wrap__content-area">
    <div class="notification-wrap__content-area__image"><svg></svg></div>
    <div class="notification-wrap__content-area__message-area">
      <p class="notification-wrap__content-area__message"></p>
      <!---->
    </div>
  </div>
  <div class="notification-wrap__buttons-group"><span class="notification-wrap__buttons-group__close"><svg></svg></span></div>
</div>
`;

exports[`NotificationItem renders correctly with error props 1`] = `
<div class="notification-wrap" style="background-color: rgb(255, 212, 210);">
  <div class="notification-wrap__content-area">
    <div class="notification-wrap__content-area__image"><svg></svg></div>
    <div class="notification-wrap__content-area__message-area">
      <p class="notification-wrap__content-area__message">testMessage</p>
      <!---->
    </div>
  </div>
  <div class="notification-wrap__buttons-group"><span class="notification-wrap__buttons-group__close"><svg></svg></span></div>
</div>
`;

exports[`NotificationItem renders correctly with notification props 1`] = `
<div class="notification-wrap" style="background-color: rgb(208, 227, 254);">
  <div class="notification-wrap__content-area">
    <div class="notification-wrap__content-area__image"><svg></svg></div>
    <div class="notification-wrap__content-area__message-area">
      <p class="notification-wrap__content-area__message">testMessage</p>
      <!---->
    </div>
  </div>
  <div class="notification-wrap__buttons-group"><span class="notification-wrap__buttons-group__close"><svg></svg></span></div>
</div>
`;

exports[`NotificationItem renders correctly with success props 1`] = `
<div class="notification-wrap" style="background-color: rgb(219, 241, 211);">
  <div class="notification-wrap__content-area">
    <div class="notification-wrap__content-area__image"><svg></svg></div>
    <div class="notification-wrap__content-area__message-area">
      <p class="notification-wrap__content-area__message">testMessage</p>
      <!---->
    </div>
  </div>
  <div class="notification-wrap__buttons-group"><span class="notification-wrap__buttons-group__close"><svg></svg></span></div>
</div>
`;

exports[`NotificationItem renders correctly with support link 1`] = `
<div class="notification-wrap" style="background-color: rgb(208, 227, 254);">
  <div class="notification-wrap__content-area">
    <div class="notification-wrap__content-area__image"><svg></svg></div>
    <div class="notification-wrap__content-area__message-area">
      <p class="notification-wrap__content-area__message">testMessage support rest of message</p> <a href="https://requestUrl.com" target="_blank" class="notification-wrap__content-area__link">
        https://requestUrl.com
      </a>
    </div>
  </div>
  <div class="notification-wrap__buttons-group"><span class="notification-wrap__buttons-group__close"><svg></svg></span></div>
</div>
`;

exports[`NotificationItem renders correctly with warning props 1`] = `
<div class="notification-wrap" style="background-color: rgb(252, 248, 227);">
  <div class="notification-wrap__content-area">
    <div class="notification-wrap__content-area__image"><svg></svg></div>
    <div class="notification-wrap__content-area__message-area">
      <p class="notification-wrap__content-area__message">testMessage</p>
      <!---->
    </div>
  </div>
  <div class="notification-wrap__buttons-group"><span class="notification-wrap__buttons-group__close"><svg></svg></span></div>
</div>
`;
